### ISSUE TITLE
Steam toast notifications are displaying at the center of the screen

### ISSUE THREAD

**Issue by hjpaul7** [ValveSoftware/steam-for-linux#11159]
#### Your system information

* Steam client version (build number or date): 1721173382
* Distribution (e.g. Ubuntu): Arch
* Opted into Steam client beta?: Yes
* Have you checked for system updates?: Yes
* Steam Logs: [steam-logs.tar.gz](https://github.com/user-attachments/files/16489416/steam-logs.tar.gz)

* GPU: Nvidia

#### Please describe your issue in as much detail as possible:
After latest beta, the Steam toast popup notifications while outside of BPM/Game appear dead center of the screen. I use Hyprland WM but it hasn't been updated/changed so this appears to be a steam bug?

Toasts are in the correct spot once in BPM or game.

#### Steps for reproducing this issue:

1. Get notification
2. 
3. 

![steam-notifications](https://github.com/user-attachments/assets/eea78f9c-4196-4b21-92be-1516f40caf6b)

**Comment by rurigk** [comment]
I dont think this is steam fault, read hyprland wiki section is "Window Rules"
Or report it to hyprland git, also update your hyprland

Maybe the window title or "class" changed and is detecting the notification as a normal window and you rules make all new windows to be on the center of the screen

**Comment by hjpaul7** [comment]
> I dont think this is steam fault, read hyprland wiki section is "Window Rules" Or report it to hyprland git, also update your hyprland

Hyprland hasn't been updated, Steam has. So this is probably coming from Steam as it was displaying correctly before.

> 
> Maybe the window title or "class" changed

[#11151](https://github.com/ValveSoftware/steam-for-linux/issues/11151)

**Comment by rurigk** [comment]
Yes but keep in mind that the windows give the DE, WM or Compositor a hint of its size and position The window tells i want to be this size and at this position, in wayland it's the decision of the compositor to take or not the hints of the window

In this case for me in plasma i dont have this problem because plasma decided to take the hints

I used Hyprland before and i had this type of issues with a lot of apps because Hyprland may not respect the hints or just ignore it but its kinda its nature, its a tiling WM at the end of the day

https://github.com/hyprwm/Hyprland/issues/3025

**Comment by hjpaul7** [comment]
> Yes but keep in mind that the windows give the DE, WM or Compositor a hint of its size and position The window tells i want to be this size and at this position, in wayland it's the decision of the compositor to take or not the hints of the window
> 
> In this case for me in plasma i dont have this problem because plasma decided to take the hints
> 
> I used Hyprland before and i had this type of issues with a lot of apps because Hyprland may not respect the hints or just ignore it but its kinda its nature, its a tiling WM at the end of the day
> 
> [hyprwm/Hyprland#3025](https://github.com/hyprwm/Hyprland/issues/3025)

I'll poke around in the Hyprland discord and see if anyone else is experiencing it.

**Comment by hallettj** [comment]
For others coming across this thread, for tiling window managers like hyprland you can add a window rule to your configuration to position toasts where you want them. You need to know what the window properties are to write matchers for the rule. After some experimentation I found that the `App ID` of all Steam windows is `steam`, and toasts have window titles of the form `notificationtoasts_20_desktop`, where the number in the title seems to increment for each toast.

I put this rule in my Niri config to position toasts where I want them:

```kdl
window-rule {
    match app-id="steam" title="^notificationtoasts"
    open-focused false
    default-floating-position x=8 y=8 relative-to="bottom-right"
    block-out-from "screencast"
}
```

**Comment by Eiernase** [comment]
Happening for me on KDE Plasma as well. Same behaviour as described in #9374, though changing to the Beta didn't fix anything. That thread is old, though.

**Comment by WanderingFox** [comment]
Also having this exact issue all of a sudden. Fedora 44 KDE Plasma. Toasts show up on the correct monitor, but at the height of the resolution of the monitor on the side.

That is to say my main monitor is 1440p and my side monitor is 1080p. When a toast appears, it appears 1,080 pixels down from the top of my main monitor.

**Comment by Konngitas** [comment]
> Also having this exact issue all of a sudden. Fedora 44 KDE Plasma. Toasts show up on the correct monitor, but at the height of the resolution of the monitor on the side.
> 
> That is to say my main monitor is 1440p and my side monitor is 1080p. When a toast appears, it appears 1,080 pixels down from the top of my main monitor.

Exact same setup and problem here
On Arch Linux 7.1.9-arch1-2, KDE Plasma Plasma 6.7.4

**Comment by LeoKeidran** [comment]
Same issue here, 1440p main display, 1080p secondary display.
On Linux Mint 22.3 - Cinnamon 6.6.9
